sdk-setup/src/sdk-manage

```diff
@@ -1565,13 +1565,23 @@ target_snapshot_needs_reset() {
         return 1
     fi
 
-    # TODO how to determine recent package removal?
     local most_recent_install_time=
     most_recent_install_time=$(set -o pipefail; rpm --root="$MER_TARGETS/$original" -qa \
-        --queryformat '%{installtime}\n' |sort -n |tail -n1) || return
-    local snapshot_time_s=$(date --date "$snapshot_time" +%s) || return
+        --queryformat '%{installtime}\n' | sort -n | tail -n1) || return
+
+    # The rpm database does not know about the most recent package removal, so
+    # parse the zypper install log. zypp/history looks like this:
+    # 2019-02-27 16:39:18|remove |droid-hal-mydevice-tools|0.0.6-201902271629|armv7hl|user@pc|
+    local most_recent_uninstall_time=$(set -o pipefail; \
+        tac $MER_TARGETS/$original/var/log/zypp/history \
+        |  grep -m 1 "|remove" \
+        | cut -d "|" -f 1 \
+    ) || return
+    most_recent_uninstall_time=$(date --date "$most_recent_uninstall_time" +%s) || return
 
-    (( most_recent_install_time > snapshot_time_s ))
+    # Compare values in unix seconds
+    local snapshot_time_s=$(date --date "$snapshot_time" +%s) || return
+    (( most_recent_install_time > snapshot_time_s ) || ( most_recent_uninstall_time > snapshot_time ))
```

Comment:
most_recent_uninstall_time > snapshot_time**_s**.

Also I do not believe this syntax would be correctly parsed by bash - ever tested it? :)

Either use ``(( i > s || r > s ))`` (notice quotes usage) or better stick snapshot_time_s definition to snapshot_time definition and then test each separately like in `(( most_receint_install_time > snapshot_time_s )) && return`

Reply:
Used `snapshot_time_s` everywhere and also picked up your suggestion so that it now read:
```
(( most_recent_install_time > snapshot_time_s )) && return
(( most_recent_removal_time > snapshot_time_s )) && return
```